### ISSUE TITLE
A generated test case tries to use a non-public class outside package

### ISSUE THREAD

**Issue by mdimjasevic** [randoop/randoop#128]
For the templateIt project from the [SF110 benchmarks](http://www.evosuite.org/experimental-data/sf110/) Randoop 3.0.8-prelease generates a test case along these lines:

```java
package randooped1;

import org.junit.FixMethodOrder;
import org.junit.Test;
import org.junit.runners.MethodSorters;
@FixMethodOrder(MethodSorters.NAME_ASCENDING)
public class test228Class {

  @Test
  public void test1() throws Throwable {


    org.apache.poi.hssf.usermodel.HSSFSheet hSSFSheet1 = null;
    java.util.List list3 = org.apache.poi.hssf.usermodel.HSSFDataFormat.getBuiltinFormats();
    // The following exception was thrown during execution in test generation                                                                                                                 
    try {
    org.templateit.DynamicTemplate dynamicTemplate4 = new org.templateit.DynamicTemplate("", hSSFSheet1, (int)(byte)0, (java.util.List<org.templateit.NamedStyle>)list3);
    } catch (java.lang.ArithmeticException e) {
      // Expected exception.                                                                                                                                                                  
      if (! e.getClass().getCanonicalName().equals("java.lang.ArithmeticException")) {
      }
    }
  }
}
```

When I try to compile this test case, I get an error:

```
test228Class.java:20: error: NamedStyle is not public in org.templateit; cannot be accessed from outside package
    org.templateit.DynamicTemplate dynamicTemplate4 = new org.templateit.DynamicTemplate("", hSSFSheet1, (int)(byte)0, (java.util.List<org.templateit.NamedStyle>)list3);
                                                                                                                                                     ^
1 error
```

If it helps, I did specify `org.templateit.NamedStyle` to Randoop as a class to be used in testing (I automatically generate a list of all .java files for a project under test).

Is it that Randoop didn't check that NamedStyle, which is not a public class, was used outside its package, which is not allowed in Java?

**Comment by bjkeller** [comment]
No, the inaccessible types given on the command-line are discarded.  It looks like this is an argument type to the constructor, meaning the constructor should have been discarded.
Thanks for the report, I will see if I can fix this before the next release.

**Comment by bjkeller** [comment]
The problem is that the argument is `List<NamedStyle>`, where `NamedStyle` is not public.
The reflection type `List<NamedStyle>` is public because `List<>` is.

**Comment by bjkeller** [comment]
fixed by 6a195546146679b0f72c85622a10ca3dad003e53
